Userland/Libraries/LibCore/System.cpp

```diff
@@ -245,6 +255,27 @@ ErrorOr<void> munmap(void* address, size_t size)
     return {};
 }
 
+ErrorOr<int> anon_create([[maybe_unused]] size_t size, [[maybe_unused]] int options)
+{
+    int fd = -1;
+#if defined(__serenity__)
+    fd = ::anon_create(size, options);
+#elif defined(__linux__)
+    // FIXME: Support more options on Linux.
+    auto linux_options = ((options & O_CLOEXEC) > 0) ? MFD_CLOEXEC : 0;
+    fd = memfd_create("", linux_options);
+    if (fd < 0)
+        return Error::from_errno(errno);
+    if (::ftruncate(fd, size) < 0) {
+        TRY(close(fd));
+        return Error::from_errno(errno);
+    }
```

Comment:
Calling `close()` after `ftruncate()` may clobber `errno` here, before it gets used.

Reply:
Good catch!